.github/workflows/generate-docs.yml

```diff
@@ -0,0 +1,71 @@
+name: Generate docs
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        required: true
+        description: "Documentation version to generate"
+      source-repo:
+        type: string
+        required: true
+        description: "Repository to build from"
+      source-ref:
+        type: string
+        required: true
+        description: "Git ref of the source repo"
+      output-prefix:
+        type: string
+        required: true
+        description: "Output path prefix"
+
+jobs:
+
+  generate-docs:
+    name: Generate docs
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Checkout ${{ github.event.inputs.source-repo }}
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.inputs.source-repo }}
+          ref: ${{ github.event.inputs.source-ref }}
+          path: source
+          fetch-depth: 0
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          architecture: x64
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r "$GITHUB_WORKSPACE/source/requirements.txt"
+      - name: Configure Git
+        run: |
+          git config user.name "afakebot"
+          git config user.email "afakebot@users.noreply.github.com"
+      - name: Get latest tag
+        id: get-latest-tag
+        uses: ./.github/actions/latest-stable-tag
+        with:
+          repo-path: source
+      - name: Generate the docs
+        run: |
+          version=${{ github.event.inputs.version }}
+          latestTag=${{ steps.get-latest-tag.outputs.latest-tag }}
+          stable=
+          if [ "$version" = $"latestTag" ]; then
+            stable="stable"
+          fi
+          targetBranch=update-$version-docs
+          git branch $targetBranch
+          mike deploy "$version" $stable \
+            --branch "$targetBranch" \
+            --prefix "${{ github.event.inputs.output-prefix }}" \
+            --config-file "$GITHUB_WORKSPACE/source/mkdocs.yml"
```

Comment:
I like this, but think we should use a custom `--message`. The default one is weak. 
Rather than focusing on the mike and mkdocs versions or whatever, I think we'd be well served to output the repo and commit (SHA preferable to (or augmenting) tag) that we're building the docs from.

Reply:
I didn't know it was an option! I'll look into it.